skrub/_datetime_encoder.py

```diff
@@ -181,23 +182,27 @@ def fit(self, X: ArrayLike, y=None) -> "DatetimeEncoder":
             Fitted DatetimeEncoder instance (self).
         """
         self._validate_keywords()
-        if isinstance(X, pd.DataFrame):
-            self.col_names_ = X.columns.to_list()
-        else:
-            self.col_names_ = None
-        X = check_input(X)
+        if not hasattr(X, "__dataframe_consortium_standard__"):
+            X = check_input(X)
+            X = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1])])
+        X = X.__dataframe_consortium_standard__()
+        if hasattr(X, 'collect'):
+            X = X.collect()
+        n_colums = len(X.column_names)
+        self.col_names_ = X.column_names
         # Features to extract for each column, after removing constant features
         self.features_per_column_ = {}
-        for i in range(X.shape[1]):
+        for i in range(n_colums):
             self.features_per_column_[i] = []
         # Check which columns are constant
-        for i in range(X.shape[1]):
+        for i in range(n_colums):
+            column = X.col(X.column_names[i])
             if self.extract_until is None:
-                if np.nanstd(self._extract_from_date(X[:, i], "total_time")) > 0:
+                if float(self._extract_from_date(column, "total_time").std()) > 0:
```

Comment:
in the Dataframe API, column reductions (e.g. `Column.std`) return ducktyped Scalars. If you want to use them within Python (e.g. here within an if-then statement), then you need to call `float` on them to materialise them to Python

Reply:
thanks - correction here, this can be simplified, because if we write
```python
if self._extract_from_date(column, "total_time").std():
```
then Python will implicitly call
```python
if self._extract_from_date(column, "total_time").std().__bool__():
```
which will materialise the scalar to Python anyway

Yes, in general, if there's a way to avoid working with Python scalars, that's better

But my suspicion is that, in general, for `fit` methods you'll need to materialise at some point (counter-examples welcome), whereas `transform` ones may be able stay lazy (at least, the ones that don't go via numpy)